egs/librispeech/s5/local/chain/run_chain_common.sh

```diff
@@ -3,133 +3,76 @@
-  steps/online/nnet2/extract_ivectors_online.sh --cmd "$train_cmd" --nj 30 \
-    data/${train_set}_min${min_seg_len}_hires_max2 \
-    exp/nnet3/extractor \
-    exp/nnet3/ivectors_${train_set}_min${min_seg_len} || exit 1;
-
- # combine the non-hires features for alignments/lattices
- rm -rf data/${latgen_train_set}_min${min_seg_len}
- utils/data/combine_short_segments.sh \
-    data/${latgen_train_set} $min_seg_len data/${latgen_train_set}_min${min_seg_len}
-fi
-
-train_set=${train_set}_min${min_seg_len}
-latgen_train_set=${latgen_train_set}_min${min_seg_len}
-ivector_dir=exp/nnet3/ivectors_${train_set}
-ali_dir=${ali_dir}_min${min_seg_len}
-lat_dir=${lat_dir}_min${min_seg_len}
-if [ $stage -le 13 ]; then
-  # realigning data as the segments would have changed
-  steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" data/$latgen_train_set data/lang $gmm_dir $ali_dir || exit 1;
-fi
-
-if [ $stage -le 14 ]; then
   # Get the alignments as lattices (gives the chain training more freedom).
   # use the same num-jobs as the alignments
   nj=$(cat ${ali_dir}/num_jobs) || exit 1;
-  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/$latgen_train_set \
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" ${lores_train_data_dir} \
     data/lang $gmm_dir $lat_dir
   rm $lat_dir/fsts.*.gz # save space
 fi
 
-mkdir -p $dir
-train_data_dir=data/${train_set}_hires
-if [ ! -z $max_wer ]; then
-  if [ $stage -le 15 ]; then
-    bad_utts_dir=${gmm_dir}_${train_set}_bad_utts
-    if [ ! -f $bad_utts_dir/all_info.sorted.txt ]; then
-      # This stage takes a lot of time ~7hrs, so run only if file is not available already
-      steps/cleanup/find_bad_utts.sh --cmd "$decode_cmd" --nj 405 data/$latgen_train_set data/lang $ali_dir $bad_utts_dir
-    fi
-    python local/sort_bad_utts.py --bad-utt-info-file $bad_utts_dir/all_info.sorted.txt --max-wer $max_wer --output-file $dir/wer_sorted_utts_${max_wer}wer
-    utils/copy_data_dir.sh --validate-opts "--no-wav"  data/${train_set}_hires data/${train_set}_${max_wer}wer_hires
-    utils/filter_scp.pl $dir/wer_sorted_utts_${max_wer}wer data/${train_set}_hires/feats.scp  > data/${train_set}_${max_wer}wer_hires/feats.scp
-    utils/fix_data_dir.sh data/${train_set}_${max_wer}wer_hires
+if [ $stage -le 13 ]; then
+  # Build a tree using our new topology. We know we have alignments for the
+  # speed-perturbed data (local/nnet3/run_ivector_common.sh made them), so use
+  # those.
+  if [ -f $tree_dir/final.mdl ]; then
+    echo "$0: $tree_dir/final.mdl already exists, refusing to overwrite it."
+    exit 1;
   fi
-  train_data_dir=data/${train_set}_${max_wer}wer_hires
-  # we don't realign again as the segment ids don't change
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate -1 \
+      --context-opts "--context-width=2 --central-position=1" \
+      --cmd "$train_cmd" 6000 data/${lores_train_data_dir} $lang $ali_dir $treedir
```

Comment:
Probably another mistake `data/${lores_train_data_dir}` should be `${lores_train_data_dir}`.